src/OpenTelemetry.Extensions.Enrichment/TraceEnrichmentProcessor.cs

```diff
@@ -0,0 +1,52 @@
+// <copyright file="TraceEnrichmentProcessor.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace OpenTelemetry.Extensions.Enrichment;
+
+#pragma warning disable CA1812 // Class is instantiated through dependency injection
+internal sealed class TraceEnrichmentProcessor : BaseProcessor<Activity>
+#pragma warning restore CA1812 // Class is instantiated through dependency injection
+{
+    private TraceEnricher[] traceEnrichers;
+
+    public TraceEnrichmentProcessor(IEnumerable<TraceEnricher> traceEnrichers)
+    {
+        this.traceEnrichers = traceEnrichers.ToArray();
+    }
+
+    public void AddEnricher(TraceEnricher traceEnricher)
```

Comment:
not sure of the use case.. I was (likely wrongly assuming!) hoping that the list of enrichers are fixed once the provider is built. When/why would AddEnricher be needed?

Reply:
OpenTelemetry .NET supports both DI and non-DI patterns and does not have plans to stick to any one of them, always want to support both. Correct me if I am wrong here .

Now, the enrichment framework was purely DI-based, and all enricher objects were injected via constructor by DI container. Since we are bringing it to the OpenTelemetry .NET world, we feel it is necessary to start supporting non-DI scenarios, therefore we should have API to add enricher objects one by one (as opposed to injecting all at once via constructor).

To be more specific, this allows users who do not use DI, add their enricher objects via this method `public static TracerProviderBuilder AddTraceEnricher(this TracerProviderBuilder builder, TraceEnricher enricher)`

Reply:
Even with such methods to support non-DI scenarios, its only valid in providerbuilder timeframe, not after it has done building the provider. So it should be having the full list while building provider, with no ability to add more after the provider is built.

Reply:
Thank you. I updated and simplified this code